keras_core/random/seed_generator.py

```diff
@@ -52,49 +48,56 @@ def __init__(self, seed=None, **kwargs):
         else:
             self.backend = backend
 
-        self._initial_seed = seed
         if seed is None:
-            seed = make_default_seed()
-
-        if not isinstance(seed, int):
-            raise ValueError(
-                "Argument `seed` must be an integer. " f"Received: seed={seed}"
-            )
+            rng, seed = backend.random.make_default_seed()
+        else:
+            rng, seed = backend.random.make_initial_seed(seed)
 
-        def seed_initializer(*args, **kwargs):
-            dtype = kwargs.get("dtype", None)
-            return self.backend.convert_to_tensor([seed, 0], dtype=dtype)
+        if backend.backend() == "tensorflow":
+            seed_dtype = "int32"
+        else:
+            seed_dtype = "uint32"
 
+        self._initial_seed = seed
+        self._rng = rng
         self.state = self.backend.Variable(
-            seed_initializer,
-            shape=(2,),
-            dtype="uint32",
+            self.backend.convert_to_tensor(seed),
+            shape=tuple(seed.shape),
+            dtype=seed_dtype,
             trainable=False,
             name="seed_generator_state",
         )
 
-    def next(self, ordered=True):
-        seed_state = self.state
-        # Use * 1 to create a copy
-        new_seed_value = seed_state.value * 1
-        if ordered:
-            increment = self.backend.convert_to_tensor(
-                np.array([0, 1]), dtype="uint32"
-            )
-            self.state.assign(seed_state + increment)
+    def next(self):
+        if backend.backend() == "torch":
```

Comment:
Can we move the torch specific code to get_next_seed_state?

Reply:
Let me elaborate why is it coded this way. 
1. In TF/JAX, when we get the next state by splitting the seed, we get two states. We assign one of these states to the state of the `SeedGenerator`, and we also pass the same state to the random functions
2. In torch, we don't have a way to control the random state explicitly. We keep passing the same gen object to random functions, and they consume and update the state of the gen implicitly. If we move this to torch backend, even then we would be required to have this condition